platform/engineering/design-docs/2020-05-26-content-validation.md

```diff
@@ -0,0 +1,148 @@
+        - [FE Tools team](#fe-tools-team)
+    - [High Level Design](#high-level-design)
+  - [Specifics](#specifics)
+    - [Detailed Design](#detailed-design)
+    - [Code Location](#code-location)
+    - [Testing Plan](#testing-plan)
+    - [Logging](#logging)
+    - [Debugging](#debugging)
+    - [Caveats](#caveats)
+    - [Security Concerns](#security-concerns)
+    - [Privacy Concerns](#privacy-concerns)
+    - [Open Questions and Risks](#open-questions-and-risks)
+    - [Work Estimates](#work-estimates)
+    - [Alternatives](#alternatives)
+      - [Highest quality, but lowest speed](#highest-quality-but-lowest-speed)
+      - [Fastest speed, but lowest quality](#fastest-speed-but-lowest-quality)
+    - [Future Work](#future-work)
+    - [Revision History](#revision-history)
+
+## Overview
+
+### Objective
+
+The objective here is to define an approach to validate the CMS content. Specifically, this design doc focuses on validating that CMS content has no accessibility errors and/or broken links.
+
+This design document is intended for front end and DevOps engineers on the Veteran-facing Services Platform (VSP) and Veteran-Facing Services (VFS) teams.
+
+### Background
+
+The `vets-website` repo currently contains [one script that builds both the content and the applications](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/stages/build/index.js). Chris V. wrote a [design doc to disentangle the content build from the application build](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/design-docs/2020-04-09-separate-content-build.md), and that work is currently in progress.
+
+Right now, content validation consists of broken link checking and accessibility checking.
+
+##### Broken Link Checking
+
+The `check-broken-link` plugin does the following:
+
+1. Loop through all of the HTML files in the Metalsmith pipeline
+2. Extract all of the broken `href`/`src` values using the `getBrokenLinks` helper
+3. Formats the result into useful console output
+4. Breaks the build on production, and logs the output on lower environments
+
+##### Accessibility Checking
+
+
+
+#### Pain points
+
+##### Content Writers
+
+Content writers want to quickly draft, publish, and deploy content. However, `vets-website` is only deployed to production once per day, and `vets-website` deploys to staging take 20+ minutes. To allow content writers to move faster, there is a partial deploy, which only includes static page changes (`vagov-content` and Drupal).
+
+Unfortunately, the partial deploys have no content validation. That means that content writers don't have a proactive reactive way to learn if their content includes broken links or accessibility errors.
+
+##### Users
+
+Content-only deploys create an opportunity for end users to encounter broken links and/or accessibility errors from CMS content that wasn't validated before being deployed.
+
+##### FE Tools team
```

Comment:
The scope is much broader here. This affects all teams across all contracts who are trying to ship code on the VA.gov front end, not just the FE Tools team.